SingularityService/src/main/java/com/hubspot/singularity/hooks/LoadBalancerClientImpl.java

```diff
@@ -68,6 +72,40 @@ public LoadBalancerClientImpl(SingularityConfiguration configuration, ObjectMapp
     this.mesosProtosUtils = mesosProtosUtils;
   }
 
+  private String getStateUriFromRequestUri(){
+    return loadBalancerUri.replace("request", "state");
+  }
+
+  private String getLoadBalancerStateUri(String loadBalancerRequestId){
+    return String.format(OPERATION_URI, getStateUriFromRequestUri(), loadBalancerRequestId);
+  }
+
+  private Optional<BaragonServiceState> getBaragonServiceStateForLoadBalancerRequest(String loadBalancerRequestId) {
+    final String loadBalancerStateUri = getLoadBalancerStateUri(loadBalancerRequestId);
+    final BoundRequestBuilder requestBuilder = httpClient.prepareGet(loadBalancerStateUri);
+    final Request request = requestBuilder.build();
+    try {
+      ListenableFuture<Response> future = httpClient.executeRequest(request);
+      Response response = future.get(loadBalancerTimeoutMillis, TimeUnit.MILLISECONDS);
+      if (!JavaUtils.isHttpSuccess(response.getStatusCode())) {
+        LOG.info(String.format("Response status code %s", response.getStatusCode()));
+      }
+      return objectMapper.readValue(response.getResponseBodyAsBytes(), new TypeReference<BaragonServiceState>() {});
+    } catch (IOException | InterruptedException | ExecutionException | TimeoutException e) {
+      e.printStackTrace();
+      return Optional.absent();
```

Comment:
I'm wondering if we should propagate this exception rather than swallowing it and returning absent here. I can see a case where Baragon is down/times out/etc and we then try to re-add upstreams it already has because this returned that the state was not present. Maybe we want to catch this exception in the actual sync poller since the poller will have more context about the appropriate action to take when that happens